commands/user_e2e_test.go

```diff
@@ -136,3 +136,81 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
 		s.Equal("Unable to find user '"+emailArg+"'", printer.GetErrorLines()[0])
 	})
 }
+
+func (s *MmctlE2ETestSuite) TestCreateUserCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	s.RunForAllClients("Should not create a user w/o username", func(c client.Client) {
+		printer.Clean()
+		cmd := &cobra.Command{}
+		cmd.Flags().String("password", "somepass", "")
+		cmd.Flags().String("email", s.th.GenerateTestEmail(), "")
+
+		err := userCreateCmdF(c, cmd, []string{})
+		s.EqualError(err, "Username is required: flag accessed but not defined: username")
+		s.Require().Empty(printer.GetLines())
+	})
+
+	s.RunForAllClients("Should not create a user w/o email", func(c client.Client) {
+		printer.Clean()
+		cmd := &cobra.Command{}
+		cmd.Flags().String("username", model.NewId(), "")
+		cmd.Flags().String("password", "somepass", "")
+
+		err := userCreateCmdF(c, cmd, []string{})
+		s.EqualError(err, "Email is required: flag accessed but not defined: email")
+		s.Require().Empty(printer.GetLines())
+	})
+
+	s.RunForAllClients("Should not create a user w/o password", func(c client.Client) {
+		printer.Clean()
+		cmd := &cobra.Command{}
+		cmd.Flags().String("username", model.NewId(), "")
+		cmd.Flags().String("email", s.th.GenerateTestEmail(), "")
+
+		err := userCreateCmdF(c, cmd, []string{})
+		s.EqualError(err, "Password is required: flag accessed but not defined: password")
+		s.Require().Empty(printer.GetLines())
+	})
+
+	s.Run("Should not create a system_admin user w/o correct privileges", func() {
+		printer.Clean()
+		cmd := &cobra.Command{}
+		cmd.Flags().String("username", model.NewId(), "")
+		cmd.Flags().String("email", s.th.GenerateTestEmail(), "")
+		cmd.Flags().String("password", "password", "")
+		cmd.Flags().Bool("system_admin", true, "")
+
+		err := userCreateCmdF(s.th.Client, cmd, []string{})
+		s.EqualError(err, "Unable to update user roles. Error: : You do not have the appropriate permissions., ")
+		s.Require().Empty(printer.GetLines())
+	})
+
+	s.RunForSystemAdminAndLocal("Should create new system_admin user given required params", func(c client.Client) {
+		printer.Clean()
+		cmd := &cobra.Command{}
+		cmd.Flags().String("username", model.NewId(), "")
+		cmd.Flags().String("email", s.th.GenerateTestEmail(), "")
+		cmd.Flags().String("password", "somepass", "")
+		cmd.Flags().Bool("system_admin", true, "")
+
+		err := userCreateCmdF(s.th.SystemAdminClient, cmd, []string{})
+		s.Require().Nil(err)
+		s.Len(printer.GetLines(), 1)
```

Comment:
Could we test at the end of the test cases that the users actually exists or not through the app layer? I'm thinking on basically the same thing that you have in the next test case but in all of them, testing both user existence and the properties that we've set.

In this case for example, it would be something like:

```go
// username and email need to be stored as vars
user, err = s.th.App.GetUserByEmail(email)
s.Equal(username, user.Username)
s.Equal(email, user.Email)
```

and for the cases where the user is not created, we can try to retrieve it with `s.th.GetUserByUsername` or `ByEmail` and assert that it doesn't exist

Reply:
Thanks for pointing this out! I've added extra assertions to all the tests in line with your suggestion.